openweather/src/main/java/pages/PlaceOrderPage.java

```diff
@@ -108,50 +105,57 @@ public PlaceOrderPage(WebDriver driver) {
         this.driver = driver;
     }
 
-    public PlaceOrderPage enterSearch(String searchStr, String searchBy) {
-        waitForPageLoaded();
-        searchTxt.click();
-        searchByPopUp.findElement(By.xpath("//button[text()='" + searchBy + "']")).click();
-        searchTxt.click();
-        searchTxt.sendKeys(searchStr);
-        sleep(2);
-        searchTxt.sendKeys(Keys.DOWN, Keys.RETURN);
-
-        return this;
-    }
-
-    public PlaceOrderPage clickAddLocationButton() {
-        new WebDriverWait(driver, TIMEOUT_IN_SECONDS)
-                .until(ExpectedConditions.visibilityOf(addLocationBtn));
-        addLocationBtn.click();
+    public PlaceOrderPage addLocation(Location location) {
+        if (location != null) {
+            waitForPageLoaded();
+            searchTxt.click();
+            switch (location.searchBy) {
+                case "By location":
+                    searchByPopUp.findElement(Locator.buttonContainText(location.searchBy)).click();
+                    searchTxt.click();
+                    searchTxt.sendKeys(location.locationOrImportFile);
+                    WaitForAction.sleep(2);
+                    searchTxt.sendKeys(Keys.DOWN, Keys.RETURN);
+                    break;
+                case "import":
+                    //implement later
+                default:
+                    //implement later
+                    break;
+            }
+            new WebDriverWait(driver, TIMEOUT_IN_SECONDS)
+                    .until(ExpectedConditions.visibilityOf(addLocationBtn));
+            addLocationBtn.click();
+        }
 
         return this;
     }
 
-    public PlaceOrderPage selectTimePeriod(String fromYear, String fromMonth, String fromDate,
-                                           String toYear, String toMonth, String toDate) {
-        fromDateInput.click();
-        fromDateInput.findElement(By.xpath(PRE_TEXT_1 + fromYear + SUF_TEXT_1)).click();
-        fromDateInput.findElement(By.xpath(PRE_TEXT_1 + fromMonth + SUF_TEXT_1)).click();
-        fromDateInput.findElement(By.xpath(PRE_TEXT_1 + fromDate + SUF_TEXT_1)).click();
-        toDateInput.click();
-        toDateInput.findElement(By.xpath(PRE_TEXT_1 + toYear + SUF_TEXT_1)).click();
-        toDateInput.findElement(By.xpath(PRE_TEXT_1 + toMonth + SUF_TEXT_1)).click();
-        toDateInput.findElement(By.xpath("//div[@class='to date-input']"
-                + PRE_TEXT_1 + toDate + SUF_TEXT_1)).click();
+    public PlaceOrderPage selectTimePeriod(Date fromDate, Date toDate) {
+        if (fromDate != null) {
```

Comment:
same as the above comment, I prefer to fail fast if the value is unexpected at runtime